### PR TITLE
Add decision-record

### PR DESCRIPTION
**Pull request by albertmink**
The Markdown documents provide information of design decision of the `versioning` of JSON files in ABAP file formats.
It contains the considered alternative options and pros/cons we discussed.
It contains discussions from #53

It also includes a index.md as a landing page for further design decision documents

### CODE REVIEW

File: doc/decision-log/0001-json-version.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+JSON schema:
+* Declare field `version` as `const` located on root level
+1. Name `ABCD-v1.json` or `ABCD-v.1.json` for the first version of the JSON schema for ABAP object type `ABCD`
+2. Store all versions in `ABCD.json` and resolve the version inside by JSON schema logic
+
+
+## Decision Outcome
+
+In the context of ABAP, `version` ambiguous as it suggests 'in-/active' versions.
```
Comment (wurzka) [inline doc/decision-log/0001-json-version.md:34]:
```suggestion
In the context of ABAP, `version` is ambiguous as it suggests 'in-/active' versions.
```

File: doc/decision-log/0001-json-version.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+— | Instance | Schema
+:--- | :--- | :---
+File name | name.ABCD-v1.json | ABCD-v1.json
+
+* Bad, because moving to the next version does rename the JSON data file
+* Bad, as the JSON data itself should know about the version
+
+### [option 2]
+
+As decides, but without field `formatVersion` or `version`.
```
Comment (schneidermic0) [inline doc/decision-log/0001-json-version.md:73]: Sorry, I don't understand this sentence :(

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks for documenting the decision. I have some added some comments to the decision about versioning

File: doc/decision-log/0001-json-version.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+* JSON data does not invalidate with URI changes in `$schema`, provoked by any change of folders or filenames
+* JSON data still decodes its version by `formatVersion`
+* Ever version of a JSON schema corresponds to a single file
+* Name `ABCD.json` is still possible, for example for a generated JSON schema that encodes all versions
+* The type `const` suits the purpose of validation of a single value (preferred over `enum`)
+
+### Negative Consequences
+
+* Build in validation and annotation, of JSON data, in editors like eclipse or VS Code is based on `$schema` and is not accessible out-of-the box
```
Comment (wurzka) [inline doc/decision-log/0001-json-version.md:52]:
```suggestion
* Build in validation and annotation of JSON data in editors like Eclipse or VS Code is based on `$schema` and is not accessible out-of-the box
```

**Review by wurzka — CHANGES_REQUESTED** [review]
Soma small remarks

File: doc/decision-log/0001-json-version.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Considered Options
+JSON data:
+1. Indicate JSON version by `$schema`
+1. Indicate JSON version by `$schema` and `version`
+1. Indicate JSON version by `version`
+
+JSON schema:
+* Declare field `version` as `const` located on root level
+1. Name `ABCD-v1.json` or `ABCD-v.1.json` for the first version of the JSON schema for ABAP object type `ABCD`
```
[the remark this answers is not in the document]
Comment (albertmink) [reply doc/decision-log/0001-json-version.md]: Jap, removed the irrelevant one.

File: doc/decision-log/0001-json-version.md (hunk 2 above)
Comment (albertmink) [reply doc/decision-log/0001-json-version.md]: As decided, ....
@schneidermic0 Does this make sense?

**Comment by albertmink** [comment]
Thanks for the review. I adopted the document to the suggestions and just re-requested your review.

**Review by schneidermic0 — APPROVED** [review]
LGTM

**Review by wurzka — APPROVED** [review]
LGTM

**Pull request merged**